### ISSUE TITLE
When Consul is down, cf push causes a CLI panic

### ISSUE THREAD

**Issue by njbennett** [cloudfoundry/cli#1351]
## Command  
cf push node3

## What occurred
After I deployed cf and then stopped consul, I attempted to push an app. Received the following error output.

```
		Error
		runtime error: invalid memory address or nil pointer dereference

		Stack Trace
			goroutine 1 [running]:
	code.cloudfoundry.org/cli/util/panichandler.HandlePanic()
		/Users/worker/concourse/volumes/live/7b67f0b0-6f6d-4a1f-740c-1e12c992ccbc/volume/gopath/src/code.cloudfoundry.org/cli/util/panichandler/handler.go:19 +0x8a
	panic(0x18c8aa0, 0x1fab2c0)
		/usr/local/Cellar/go/1.10/libexec/src/runtime/panic.go:505 +0x229
	code.cloudfoundry.org/cli/command/translatableerror.UploadFailedError.Translate(0x0, 0x0, 0xc420508000, 0x17f0601, 0x25b26e8)
		/Users/worker/concourse/volumes/live/7b67f0b0-6f6d-4a1f-740c-1e12c992ccbc/volume/gopath/src/code.cloudfoundry.org/cli/command/translatableerror/upload_failed_error.go:16 +0x1d2
	code.cloudfoundry.org/cli/util/ui.(*UI).DisplayError(0xc4203ca500, 0x1abd160, 0xc420166d20)
		/Users/worker/concourse/volumes/live/7b67f0b0-6f6d-4a1f-740c-1e12c992ccbc/volume/gopath/src/code.cloudfoundry.org/cli/util/ui/ui.go:174 +0x8b
	main.handleError(0x1abbc80, 0xc420084220, 0x1ac3180, 0xc4203ca500, 0x1abbc80, 0xc420084220)
		/Users/worker/concourse/volumes/live/7b67f0b0-6f6d-4a1f-740c-1e12c992ccbc/volume/gopath/src/code.cloudfoundry.org/cli/main.go:257 +0x9a
	main.executionWrapper(0x2570640, 0x1fd1a18, 0xc4201a28a0, 0x0, 0x2, 0x0, 0x0)
		/Users/worker/concourse/volumes/live/7b67f0b0-6f6d-4a1f-740c-1e12c992ccbc/volume/gopath/src/code.cloudfoundry.org/cli/main.go:216 +0x25e
	code.cloudfoundry.org/cli/vendor/github.com/jessevdk/go-flags.(*Parser).ParseArgs(0xc420304060, 0xc4200300d0, 0x2, 0x2, 0x10, 0xc420508400, 0x0, 0x1fc26a0, 0x19ee071)
		/Users/worker/concourse/volumes/live/7b67f0b0-6f6d-4a1f-740c-1e12c992ccbc/volume/gopath/src/code.cloudfoundry.org/cli/vendor/github.com/jessevdk/go-flags/parser.go:314 +0x7c7
	main.parse(0xc4200300d0, 0x2, 0x2, 0xc420046118)
		/Users/worker/concourse/volumes/live/7b67f0b0-6f6d-4a1f-740c-1e12c992ccbc/volume/gopath/src/code.cloudfoundry.org/cli/main.go:52 +0xad
	main.main()
		/Users/worker/concourse/volumes/live/7b67f0b0-6f6d-4a1f-740c-1e12c992ccbc/volume/gopath/src/code.cloudfoundry.org/cli/main.go:43 +0x82

	goroutine 5 [syscall]:
	os/signal.signal_recv(0x0)
		/usr/local/Cellar/go/1.10/libexec/src/runtime/sigqueue.go:139 +0xa7
	os/signal.loop()
		/usr/local/Cellar/go/1.10/libexec/src/os/signal/signal_unix.go:22 +0x22
	created by os/signal.init.0
		/usr/local/Cellar/go/1.10/libexec/src/os/signal/signal_unix.go:28 +0x41

	goroutine 573 [select]:
	net/http.(*persistConn).writeLoop(0xc420311680)
		/usr/local/Cellar/go/1.10/libexec/src/net/http/transport.go:1822 +0x14b
	created by net/http.(*Transport).dialConn
		/usr/local/Cellar/go/1.10/libexec/src/net/http/transport.go:1238 +0x97f

	goroutine 572 [IO wait]:
	internal/poll.runtime_pollWait(0x25a8ea0, 0x72, 0xc42007f760)
		/usr/local/Cellar/go/1.10/libexec/src/runtime/netpoll.go:173 +0x57
	internal/poll.(*pollDesc).wait(0xc4201b8698, 0x72, 0xffffffffffffff00, 0x1abdce0, 0x1ef0518)
		/usr/local/Cellar/go/1.10/libexec/src/internal/poll/fd_poll_runtime.go:85 +0x9b
	internal/poll.(*pollDesc).waitRead(0xc4201b8698, 0xc420668000, 0x800, 0x800)
		/usr/local/Cellar/go/1.10/libexec/src/internal/poll/fd_poll_runtime.go:90 +0x3d
	internal/poll.(*FD).Read(0xc4201b8680, 0xc420668000, 0x800, 0x800, 0x0, 0x0, 0x0)
		/usr/local/Cellar/go/1.10/libexec/src/internal/poll/fd_unix.go:157 +0x1dc
	net.(*netFD).Read(0xc4201b8680, 0xc420668000, 0x800, 0x800, 0x63, 0xc420204d4e, 0x2)
		/usr/local/Cellar/go/1.10/libexec/src/net/fd_unix.go:202 +0x4f
	net.(*conn).Read(0xc4201ac020, 0xc420668000, 0x800, 0x800, 0x0, 0x0, 0x0)
		/usr/local/Cellar/go/1.10/libexec/src/net/net.go:176 +0x6a
	crypto/tls.(*block).readFromUntil(0xc4204b8000, 0x25707a0, 0xc4201ac020, 0x5, 0xc4201ac020, 0xc42007fa90)
		/usr/local/Cellar/go/1.10/libexec/src/crypto/tls/conn.go:493 +0x96
	crypto/tls.(*Conn).readRecord(0xc4200c4380, 0x1a32317, 0xc4200c44a0, 0x1053720)
		/usr/local/Cellar/go/1.10/libexec/src/crypto/tls/conn.go:595 +0xe0
	crypto/tls.(*Conn).Read(0xc4200c4380, 0xc420301000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
		/usr/local/Cellar/go/1.10/libexec/src/crypto/tls/conn.go:1156 +0x100
	net/http.(*persistConn).Read(0xc420311680, 0xc420301000, 0x1000, 0x1000, 0xc42007fb98, 0x10052f5, 0xc420095020)
		/usr/local/Cellar/go/1.10/libexec/src/net/http/transport.go:1453 +0x136
	bufio.(*Reader).fill(0xc4200a0540)
		/usr/local/Cellar/go/1.10/libexec/src/bufio/bufio.go:100 +0x11e
	bufio.(*Reader).Peek(0xc4200a0540, 0x1, 0x0, 0x0, 0x0, 0xc420047260, 0x0)
		/usr/local/Cellar/go/1.10/libexec/src/bufio/bufio.go:132 +0x3a
	net/http.(*persistConn).readLoop(0xc420311680)
		/usr/local/Cellar/go/1.10/libexec/src/net/http/transport.go:1601 +0x185
	created by net/http.(*Transport).dialConn
		/usr/local/Cellar/go/1.10/libexec/src/net/http/transport.go:1237 +0x95a

	goroutine 675 [select]:
	code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb%2ev1.(*ProgressBar).refresher(0xc42045c000)
		/Users/worker/concourse/volumes/live/7b67f0b0-6f6d-4a1f-740c-1e12c992ccbc/volume/gopath/src/code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb.v1/pb.go:462 +0xf5
	created by code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb%2ev1.(*ProgressBar).Start
		/Users/worker/concourse/volumes/live/7b67f0b0-6f6d-4a1f-740c-1e12c992ccbc/volume/gopath/src/code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb.v1/pb.go:123 +0xc0

	goroutine 878 [select, locked to thread]:
	runtime.gopark(0x1a32278, 0x0, 0x19ec7cf, 0x6, 0x18, 0x1)
		/usr/local/Cellar/go/1.10/libexec/src/runtime/proc.go:291 +0x11a
	runtime.selectgo(0xc4202cf750, 0xc4205423c0)
		/usr/local/Cellar/go/1.10/libexec/src/runtime/select.go:392 +0xe50
	runtime.ensureSigM.func1()
		/usr/local/Cellar/go/1.10/libexec/src/runtime/signal_unix.go:549 +0x1c6
	runtime.goexit()
		/usr/local/Cellar/go/1.10/libexec/src/runtime/asm_amd64.s:2361 +0x1

	goroutine 879 [chan receive]:
	code.cloudfoundry.org/cli/util/configv3.catchSignal(0xc4205e80c0, 0xc4205f00f0, 0x27)
		/Users/worker/concourse/volumes/live/7b67f0b0-6f6d-4a1f-740c-1e12c992ccbc/volume/gopath/src/code.cloudfoundry.org/cli/util/configv3/write_config.go:56 +0x39
	created by code.cloudfoundry.org/cli/util/configv3.WriteConfig
		/Users/worker/concourse/volumes/live/7b67f0b0-6f6d-4a1f-740c-1e12c992ccbc/volume/gopath/src/code.cloudfoundry.org/cli/util/configv3/write_config.go:40 +0x2da

	goroutine 866 [select]:
	code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb%2ev1.(*ProgressBar).refresher(0xc4204e06e0)
		/Users/worker/concourse/volumes/live/7b67f0b0-6f6d-4a1f-740c-1e12c992ccbc/volume/gopath/src/code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb.v1/pb.go:462 +0xf5
	created by code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb%2ev1.(*ProgressBar).Start
		/Users/worker/concourse/volumes/live/7b67f0b0-6f6d-4a1f-740c-1e12c992ccbc/volume/gopath/src/code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb.v1/pb.go:123 +0xc0

	goroutine 685 [select]:
	code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb%2ev1.(*ProgressBar).refresher(0xc4204e0b00)
		/Users/worker/concourse/volumes/live/7b67f0b0-6f6d-4a1f-740c-1e12c992ccbc/volume/gopath/src/code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb.v1/pb.go:462 +0xf5
	created by code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb%2ev1.(*ProgressBar).Start
		/Users/worker/concourse/volumes/live/7b67f0b0-6f6d-4a1f-740c-1e12c992ccbc/volume/gopath/src/code.cloudfoundry.org/cli/vendor/gopkg.in/cheggaaa/pb.v1/pb.go:123 +0xc0
```

## What you expected to occur
`cf push` to fail with an normal error response, not a stack trace

## CLI Version  
6.35.2+88a03e995.2018-03-15

## CC API Endpoint Version  
2.105.0

## CF Trace
RESPONSE: [2018-03-22T16:22:43-07:00]
HTTP/1.1 500 Internal Server Error
Alt-Svc: clear
Content-Length: 99
Content-Type: application/json;charset=utf-8
Date: Thu, 22 Mar 2018 23:22:43 GMT
Server: nginx
Via: 1.1 google
X-Content-Type-Options: nosniff
X-Vcap-Request-Id: 921b3f30-2a33-46df-4aca-c08d6e7ea10a::a6d8e066-c24a-42ed-bb10-891bdcc6432c
{
  "code": 10001,
  "description": "An unknown error occurred.",
  "error_code": "UnknownError"
}

Retrying upload due to an error...
Uploading files...
 0 B / 623 B [---------------------------------------------------------------------------------------------------------------------------------------]   0.00%REQUEST: [2018-03-22T16:22:43-07:00]
PUT /v2/apps/739580b7-e468-4da1-83b0-dca904c2f104/bits?async=true HTTP/1.1
Host: api.hagrid.cf-app.com
Accept: application/json
Authorization: [PRIVATE DATA HIDDEN]
Content-Type: multipart/form-data; boundary=aecb5d740af462e254aa71ed7508f3e0fe91e7b38c24ea6b92c6acb3c873
User-Agent: cf/6.35.2+88a03e995.2018-03-15 (go1.10; amd64 darwin)
[multipart/form-data Content Hidden]

 623 B / 623 B [=====================================================================================================================================] 100.00%RESPONSE: [2018-03-22T16:22:43-07:00]
HTTP/1.1 500 Internal Server Error
Alt-Svc: clear
Content-Length: 99
Content-Type: application/json;charset=utf-8
Date: Thu, 22 Mar 2018 23:22:43 GMT
Server: nginx
Via: 1.1 google
X-Content-Type-Options: nosniff
X-Vcap-Request-Id: da47b6cc-97c1-43d0-65bd-d45f454f1763::ca9d064b-8e30-4109-b4ce-f5783bfc0b16
{
  "code": 10001,
  "description": "An unknown error occurred.",
  "error_code": "UnknownError"
}

## Platform & Shell Details  
Mac OS X 10.13.3
iTerm 3.1.5.beta.2

### Environment deploy command
`bosh deploy ../cf-deployment/cf-deployment.yml -o ../cf-deployment/operations/scale-to-one-az.yml -o ../cf-deployment/operations/experimental/fast-deploy-with-downtime-and-danger.yml -o ../cf-deployment/operations/use-compiled-releases.yml -v system_domain=hagrid.cf-app.com --vars-store vars-store/deployment-vars.yml`

followed by
`bosh stop consul`

**Comment by XenoPhex** [comment]
Dupe of #1340.

**Comment by XenoPhex** [comment]
Fixed in 6.35.*.
